rust/parser/literal.rs

```diff
@@ -107,12 +120,18 @@ fn visit_date_fragment(node: Node<'_>) -> DateFragment {
 
 fn visit_time(node: Node<'_>) -> TimeFragment {
     debug_assert_eq!(node.as_rule(), Rule::time);
-    let children = node.into_children().collect::<Vec<_>>();
-    let (hour, minute, second) = (
-        children[0].as_str().to_owned(),
-        children[1].as_str().to_owned(),
-        children.get(2).map(|node| node.as_str().to_owned()),
-    );
-    let second_fraction = children.get(3).map(|node| node.as_str().to_owned());
+    let mut children = node.into_children();
+    let hour = children.consume_expected(Rule::hour).as_str().to_owned();
+    let minute = children.consume_expected(Rule::minute).as_str().to_owned();
+    let second = if children.peek().is_some() {
+        Some(children.consume_expected(Rule::second).as_str().to_owned())
+    } else {
+        None
+    };
+    let second_fraction = if children.peek().is_some() {
+        Some(children.consume_expected(Rule::second_fraction).as_str().to_owned())
+    } else {
+        None
+    };
     TimeFragment { hour, minute, second, second_fraction }
```

Comment:
One thing that's missing in these visitors is the assertion that the `children` iterator is fully exhausted. (`debug_assert_eq!(children.try_consume_any(), None)`)